### PR TITLE
add ConnectionNofication component

### PR DESCRIPTION
**Pull request by raymondz1**
## What this PR does / why we need it

improved the user experience for the connect button. It can show error message and loading state according to the response of api call. 

## Which issue(s) this PR fixes

Fixes: #

## Special notes for your reviewer

Screen Shot 2022-07-13 at 4 01 02 PM

Screen Shot 2022-07-14 at 11 10 49 AM

### CODE REVIEW

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/wizard-basic/management-credential-step/ManagementCredentials.tsx [hunk 1]
```diff
@@ -76,15 +77,18 @@ function ManagementCredentials(props: Partial<StepProps>) {
...
+            setMessage('Connecting to Azure');
             await AzureService.setAzureEndpoint(params);
-            setConnection(true);
+            setStatus(CONNECTION_STATUS.CONNECTED);
+            setMessage('Connected to Azure');
             retrieveRegions();
-        } catch (e) {
-            console.log('Error');
+        } catch (err: any) {
+            setMessage(err.body.message);
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/wizard-basic/management-credential-step/ManagementCredentials.tsx:87]: Please add some curated text to the beginning of the error message like: Unable to connect to Azure: ...error message here...

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/docker/wizard-basic/mc-prerequisite-step/McPrerequisiteStep.tsx [hunk 2]
```diff
@@ -11,18 +10,22 @@ import { CriService } from '../../../../../swagger-api';
...
         try {
             await CriService.getContainerRuntimeInfo();
-            setConnection(true);
+            setStatus(CONNECTION_STATUS.CONNECTED);
+            setMessage('Running Docker daemon');
         } catch (err: any) {
-            setErrorMessage(err.body.message);
-            setConnection(false);
+            setStatus(CONNECTION_STATUS.ERROR);
+            setMessage(err.body.message);
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/docker/wizard-basic/mc-prerequisite-step/McPrerequisiteStep.tsx:28]: same here

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentials.tsx [hunk 3]
```diff
@@ -101,13 +104,20 @@ function ManagementCredentials(props: Partial<StepProps>) {
...
-            setConnection(true);
-        });
+        try {
+            setStatus(CONNECTION_STATUS.CONNECTING);
+            setMessage('Connecting to AWS');
+            await AwsService.setAwsEndpoint(params);
+            setStatus(CONNECTION_STATUS.CONNECTED);
+            setMessage('Connected to AWS');
+        } catch (err: any) {
+            setStatus(CONNECTION_STATUS.ERROR);
```
Comment (miclettej) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentials.tsx:114]: same as below

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/wizard-basic/management-credential-step/ManagementCredentials.tsx (hunk 1 above)
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/wizard-basic/management-credential-step/ManagementCredentials.tsx]: fixed.

**Review by miclettej — APPROVED** [review]

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentials.tsx [hunk 4]
```diff
@@ -43,6 +44,9 @@ enum CREDENTIAL_TYPE {
 function ManagementCredentials(props: Partial<StepProps>) {
     const { handleValueChange, currentStep, goToStep, submitForm } = props;
     const { awsState } = useContext(AwsStore);
+    const [status, setStatus] = useState<CONNECTION_STATUS>(CONNECTION_STATUS.DISCONNECTED);
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentials.tsx:47]: nit: I'd prefer `connectionStatus` over just `status`

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentials.tsx [hunk 5]
```diff
@@ -190,11 +200,15 @@ function ManagementCredentials(props: Partial<StepProps>) {
...
-                <div cds-layout="p-t:lg">
-                    <CdsButton onClick={handleConnect} disabled={connected || !awsState[STORE_SECTION_FORM].REGION}>
+                <div cds-layout="p-t:lg" className="aws-button-container">
+                    <CdsButton
+                        onClick={handleConnect}
+                        disabled={status === CONNECTION_STATUS.CONNECTED || !awsState[STORE_SECTION_FORM].REGION}
+                    >
                         <CdsIcon shape="connect" size="md"></CdsIcon>
-                        {connected ? 'CONNECTED' : 'CONNECT'}
+                        {status === CONNECTION_STATUS.CONNECTED ? 'CONNECTED' : 'CONNECT'}
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentials.tsx:209]: I thought our standard was to deactivate the button when connected (and not change the text)

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentials.tsx (hunk 4 above)
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentials.tsx]: fixed.

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentials.tsx (hunk 5 above)
Comment (raymondz1) [reply cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentials.tsx]: fixed.

File: cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/wizard-basic/management-credential-step/ManagementCredentials.tsx [hunk 6]
```diff
@@ -76,15 +77,18 @@ function ManagementCredentials(props: Partial<StepProps>) {
...
+            setMessage('Connecting to Azure');
             await AzureService.setAzureEndpoint(params);
-            setConnection(true);
+            setConnectionStatus(CONNECTION_STATUS.CONNECTED);
+            setMessage('Connected to Azure');
             retrieveRegions();
-        } catch (e) {
-            console.log('Error');
+        } catch (err: any) {
+            setMessage(`Unable to connect to Azure: ${err.body.message}`);
```
Comment (swalner-vmware) [inline cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/wizard-basic/management-credential-step/ManagementCredentials.tsx:87]: on the error, I think (if I recall correctly) the connection status should be set to ERROR.

**Pull request merged**